app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_posts_counter
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] [OPTIONAL] it would be better if you used `increment` to update the post counter, it's easier and street forward 
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```